src/app/shared/wysiwyg/prosemirror.markdown.editor.ts

```diff
@@ -20,29 +20,23 @@ export class ProseMirrorImage {
 export class ProseMirrorMarkdownEditor {
 
   view: MenuBarEditorView;
-  savedState: EditorState;
 
   constructor(private el: ElementRef,
               private value: string,
               private images: ProseMirrorImage[],
               private setModel: Function,
               private promptForLink: Function) {
-    this.savedState = EditorState.create(this.stateConfig());
-    this.view = new MenuBarEditorView(el.nativeElement, this.viewProps(this.savedState));
+    let state = EditorState.create(this.stateConfig());
+    this.view = new MenuBarEditorView(el.nativeElement, this.viewProps(state));
   }
 
-  update(images: ProseMirrorImage[]) {
-    this.images = images;
-    let state = this.view.editor.state.reconfigure(this.stateConfig());
-    this.view.update(this.viewProps(state));
-  }
-
-  resetEditor() {
-    this.view.updateState(this.savedState);
-  }
-
-  setSavedState() {
-    this.savedState = this.view.editor.state;
+  update(value: string, images?: ProseMirrorImage[]) {
+    if (value !== this.value || images) {
+      this.value = value;
+      this.images = images || this.images;
```

Comment:
I see, the ngOnChanges logic only passes in the `images` when they have been changed, so no need to compare them.